certbot-ci/launch-integration-tests.py

```diff
@@ -0,0 +1,232 @@
+        - generate the pytest command
+        - create a temporary workspace and the persistent GIT repositories space
+        - execute the pytest command in the temporary workspace
+        - launch the coverage if needed
+    :param cli_args: The arguments given by the CLI
+    :return: 0 if the integration tests execution is considered as successful, 1 otherwise
+    :rtype: int
+    """
+    main_parser = create_parser()
+    args = main_parser.parse_args(cli_args)
+
+    # Check for runtime compatiblity: some tools are required to be available in PATH
+    try:
+        subprocess.check_output(['docker', '-v'], stderr=subprocess.STDOUT)
+    except (subprocess.CalledProcessError, OSError):
+        raise ValueError('Error: docker is required in PATH to launch the integration tests, '
+                         'but is not installed or not available for current user.')
+
+    try:
+        subprocess.check_output(['docker-compose', '-v'], stderr=subprocess.STDOUT)
+    except (subprocess.CalledProcessError, OSError):
+        raise ValueError('Error: docker-compose is required in PATH to launch the integration tests, '
+                         'but is not installed or not available for current user.')
+
+    # We will know, after the next command, how many workers need to be setup. Pytest command
+    # will contain options that rely on several pytest plugins. Theses plugins should have been
+    # installed through pip using the setup.py of certbot-ci project.
+    (command, workers) = prepare_pytest_command(args)
+
+    repositories_path = os.path.join(CURRENT_DIR, '.ci_assets/integration_tests')
+    try:
+        os.makedirs(repositories_path)
+    except OSError as error:
+        if error.errno != errno.EEXIST:
+            raise
+
+    # By calling certbot_ci_workspace, the tempfile.tempdir value of standard tempfile module is
+    # modified for the context execution time, to ensure that any temporary assets will be created
+    # under the cerbot-ci workspace.
+    with certbot_ci_workspace():
+        acme_config = {}
+        # Prepare the acme config server. Data is specific to an acme type. Module
+        # utils.acme_server will handle theses specifics.
+        if 'pebble' in args.acme_server:
+            acme_config['type'] = 'pebble'
+            acme_config['option'] = 'nonstrict' if 'nonstrict' in args.acme_server else 'strict'
+        else:
+            acme_config['type'] = 'boulder'
+            acme_config['option'] = 'v1' if 'v1' in args.acme_server else 'v2'
+        # By calling setup_acme_server we ensure that all necessary acme servers instances will be
+        # fully started once the context is entered. This runtime is reflected by the acme_xdist
+        # provided by the contextmanager.
+        with acme.setup_acme_server(acme_config, workers, repositories_path) as acme_xdist:
+            os.environ['CERTBOT_ACME_TYPE'] = args.acme_server
```

Comment:
We are passing command line arguments understood by pytest. We need something to pass other contexts. The most easiest way is to set en environment variable. And, you can also use a hook on pytest to extend its capabilities towards command line arguments, and add a new one.

It would work well for `CERTBOT_ACME_XDIST`. However it will be a problem for `CERTBOT_ACME_TYPE`, that will be used in part 3 in a area that have no direct access to the pytest context execution.

Instead of splitting in two ways the context I want to pass to pytest, I use only one, and choose what is working for both contexts.